### PR TITLE
feat(n2n): add database structure for n2n

### PR DESCRIPTION
**Pull request by Phil91**
## Description

add new database structure for network to network
add seeding for n2n
adjust POST: api/administration/identityprovider/owncompany/identityproviders

## Why

To prepare the database for the n2n changes

## Issue

N/A - Jira Issues: CPLP-3151, CPLP-3145, CPLP-3149, CPLP-3150, CPLP-3200

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/portalbackend/PortalBackend.Migrations/Migrations/20230904091939_CPLP-3165-AddN2NModels.cs [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+        }
+
+        /// <inheritdoc />
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.Sql("DROP FUNCTION \"portal\".\"LC_TRIGGER_AFTER_INSERT_COMPANYAPPLICATION\"() CASCADE;");
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.Migrations/Migrations/20230904091939_CPLP-3165-AddN2NModels.cs:238]: fails with error '42883: function portal.LC_TRIGGER_AFTER_INSERT_COMPANYAPPLICATION() does not exist' (this is due to the schema and function-names are enclosed in "..")
not sure why this is the case as it should be generated? Provious migration 'AdjustTriggers' seems to be all fine in respect to this.

File: src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs [hunk 2]
```diff
@@ -2728,8 +2842,8 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
...trigger as $LC_TRIGGER_AFTER_INSERT_CONNECTOR$\r\nBEGIN\r\n  INSERT INTO \"portal\".\"audit_connector20230803\" (\"self_description_message\", \"provider_id\", \"company_service_account_id\", \"self_description_document_id\", \"host_id\", \"date_last_changed\", \"name\", \"last_editor_id\", \"id\", \"connector_url\", \"location_id\", \"status_id\", \"type_id\", \"audit_v1id\", \"audit_v1operation_id\", \"audit_v1date_last_changed\", \"audit_v1last_editor_id\") SELECT NEW.\"self_description_message\", \r\n  NEW.\"provider_id\", \r\n  NEW.\"company_service_account_id\", \r\n  NEW.\"self_description_document_id\", \r\n  NEW.\"host_id\", \r\n  NEW.\"date_last_changed\", \r\n  NEW.\"name\", \r\n  NEW.\"last_editor_id\", \r\n  NEW.\"id\", \r\n  NEW.\"connector_url\", \r\n  NEW.\"location_id\", \r\n  NEW.\"status_id\", \r\n  NEW.\"type_id\", \r\n  gen_random_uuid(), \r\n  1, \r\n  CURRENT_TIMESTAMP, \r\n  NEW.\"last_editor_id\";\r\nRETURN NEW;\r\nEND;\r\n$LC_TRIGGER_AFTER_INSERT_CONNECTOR$ LANGUAGE plpgsql;\r\nCREATE TRIGGER LC_TRIGGER_AFTER_INSERT_CONNECTOR AFTER INSERT\r\nON \"portal\".\"connectors\"\r\nFOR EACH ROW EXECUTE PROCEDURE \"portal\".\"LC_TRIGGER_AFTER_INSERT_CONNECTOR\"();")
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs:2845]: those Annotations are unrelated to N2N so they shouldn't have changed?

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Comment by ntruchsess** [comment]
regarding the issues with trigger-annotations in context-snapshot:
I have installed ef-tools latest version 7.0.10 and when I checkout dev and add a new migration this migration is not empty but contains all trigger-function being recreated.

File: src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs (hunk 2 above)
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs:2845]: this is fixed in https://github.com/eclipse-tractusx/portal-backend/pull/254

**Comment by ntruchsess** [comment]
> regarding the issues with trigger-annotations in context-snapshot: I have installed ef-tools latest version 7.0.10 and when I checkout dev and add a new migration this migration is not empty but contains all trigger-function being recreated.

this is fixed in https://github.com/eclipse-tractusx/portal-backend/pull/254

File: src/portalbackend/PortalBackend.Migrations/PortalBackend.Migrations.csproj [hunk 3]
```diff
@@ -74,4 +72,9 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
   </ItemGroup>
+
+  <ItemGroup>
+    <Compile Remove="Migrations\20230825073236_CPLP-3165-AddN2NModels.cs" />
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.Migrations/PortalBackend.Migrations.csproj]: removed

File: src/portalbackend/PortalBackend.Migrations/Migrations/20230904091939_CPLP-3165-AddN2NModels.cs [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+                values: new object[,]
+                {
+                    { 1, "OWN" },
+                    { 2, "MANAGED" }
+                });
+
+            migrationBuilder.Sql("UPDATE portal.identity_providers SET identity_provider_type_id = 1 WHERE identity_provider_category_id = 1");
+            migrationBuilder.Sql("UPDATE portal.identity_providers SET identity_provider_type_id = 2 WHERE identity_provider_category_id != 1");
+            migrationBuilder.Sql("UPDATE portal.identity_providers SET owner_id = '2dc4249f-b5ca-4d42-bef1-7a7a950a4f87'");
+            migrationBuilder.Sql("UPDATE portal.identity_providers AS idp SET owner_id = (SELECT cip.company_id FROM portal.company_identity_providers AS cip WHERE idp.id = cip.identity_provider_id) WHERE idp.id IN (SELECT cip.identity_provider_id FROM portal.company_identity_providers AS cip);");
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20230904091939_CPLP-3165-AddN2NModels.cs]: removed the migration scripts and set ownerId to optional

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**